### DISCUSSION TITLE
agent memory如何使用变量进行memory的存储

### DISCUSSION THREAD

**Discussion by ACBBZ** [modelscope/ms-agent#407] [Q&A]
我希望使用一个变量进行history的存储，而不是本地的文件进行history的存储。
例如：
`llm_config = ...

messages = [{
    'role': 'user',
    'content': 'Hello.'
}, {
    'role': 'assistant',
    'content': 'Hi there!'
}, {
    'role': 'user',
    'content': 'Tell me a joke.'
}]

function_list = []
bot = RolePlay(function_list=function_list, llm=llm_config)

response = bot.run(messages)
`

**Comment by zzhangpurdue** [comment]
参考一下这个case：https://github.com/modelscope/modelscope-agent/blob/master/demo/demo_multi_round_agent.ipynb
